### PR TITLE
Upgrade tests from template

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ endif
 
 ifndef NOCOLOR
   GREEN  := $(shell tput -Txterm setaf 2)
-  YELLOW := $(shell tput -Txterm setaf 3)
+  YELLOW := $(shell tput -Txterm seaf 3)
```
Comment (mariux) [inline Makefile:28]: this is broken. please revert

**Review by mariux — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -119,4 +123,4 @@ DOCKER_RUN_CMD  = docker run ${DOCKER_FLAGS} ${BUILD_TOOLS_DOCKER_IMAGE}
 quiet-command = $(if ${V},${1},$(if ${2},@echo ${2} && ${1}, @${1}))
 docker-run    = $(call quiet-command,${DOCKER_RUN_CMD} ${1} | cat,"${YELLOW}[DOCKER RUN] ${GREEN}${1}${RESET}")
 go-test       = $(call quiet-command,${DOCKER_RUN_CMD} go test -v -count 1 -timeout 45m -parallel 128 ${1} | cat,"${YELLOW}[TEST] ${GREEN}${1}${RESET}")
-rm-command    = $(call quiet-command,rm -rf ${1},"${YELLOW}[CLEAN] ${GREEN}${1}${RESET}")
+rm-command    = $(call quiet-command,rm -rf ${1},"${YELLOW}[CLEAN] ${GREEN}${1}${RESET}")t
```
Comment (mariux) [inline Makefile:126]: this is broken please revert

File: test/unit-disabled/main.tf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  description = "(Optional) The AWS region in which all resources will be created."
+  type        = string
+  default     = "us-east-1"
+}
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
```
Comment (mariux) [inline test/unit-disabled/main.tf:17]:
```suggestion
      version = "~> 3.50"
```

File: test/unit-complete/main.tf [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  description = "(Optional) The AWS region in which all resources will be created."
+  type        = string
+  default     = "us-east-1"
+}
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
```
Comment (mariux) [inline test/unit-complete/main.tf:17]:
```suggestion
      version = "~> 3.50"
```

**Review by mariux — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (soerenmartius) [reply Makefile]: resolved

File: Makefile (hunk 2 above)
Comment (soerenmartius) [reply Makefile]: resolved

File: test/unit-disabled/main.tf (hunk 3 above)
Comment (soerenmartius) [reply test/unit-disabled/main.tf]: resolved

File: test/unit-complete/main.tf (hunk 4 above)
Comment (soerenmartius) [reply test/unit-complete/main.tf]: resolved

File: test/unit-complete/.terraform.lock.hcl [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# This file is maintained automatically by "terraform init".
```
Comment (i4ki) [inline test/unit-complete/.terraform.lock.hcl:1]: is this file needed to be committed?

Comment (i4ki) [reply test/unit-complete/.terraform.lock.hcl:1]: if not, check `unit-disabled/` as well.

Comment (soerenmartius) [reply test/unit-complete/.terraform.lock.hcl]: you're right, it shouldn't be part of this PR - will change

Comment (soerenmartius) [reply test/unit-complete/.terraform.lock.hcl]: removed in 985e154d

**Review by samen93 — APPROVED** [review]
